spec/rspec/core/configuration_spec.rb

```diff
@@ -36,6 +36,24 @@ module RSpec::Core
       end
     end
 
+    describe 'Corrupt examples.txt file' do
+      before do
+        RSpec.configure do |c|
+          c.example_status_persistence_file_path = "spec/examples-corrupted.txt"
+        end
+      end
+
+      it 'should not crash' do
+        Array.new(9) do |i|
+          RSpec.describe("group") { it "example #{i}" do; end}
+        end
+        expectation = expect do
+          RSpec.world.example_groups
+        end
+        expectation.not_to raise_error
+      end
+    end
```

Comment:
A few things...

1. Since the failure is in `last_run_statuses` and we already have some specs specifically for that method, I'd prefer this spec was moved to there and re-written to exercise that directly.  The specs I'm referring to are [here](https://github.com/rspec/rspec-core/blob/d0d67bf0909def9cfe35e120f82844ff59ebf57c/spec/rspec/core/configuration/only_failures_support_spec.rb#L12-L87).

2. Not raising an error is a good thing, but I think we can make the assertion even stronger; specifically that the status is set to `UNKNOWN_STATUS`.  Otherwise, a fix of `example.fetch(:status, "BWAH HA HA")` could still pass, but the rest of the code in RSpec does not understand that status.  So in your re-written spec, please assert that the status is set to `UNKNOWN_STATUS`.

3.  I think in the other spec file you can get by w/o having to write an external fixture like you did here--instead you can just pass hashes to the `simulate_persisted_examples` method w/o a `:status` key.  That helps keeps everything necessary to understand the spec visible in one place.